packages/eslint-plugin/rules/jsx-a11y-anchor-has-content.js

```diff
@@ -0,0 +1,161 @@
+/**
+ *
+ * @file Wraps eslint-plugin-jsx-a11y/anchor-has-content to allow usage with createInterpolateElement.
+ *
+ */
+
+/**
+ * External dependencies
+ *
+ * eslint-plugin-jsx-a11y/anchor-has-content: to create a wrapper around it, we need to use it
+ * elementType from jsx-ast-utils: to get the type of the node we are working with.
+ */
+const anchorRule = require( 'eslint-plugin-jsx-a11y' ).rules[
+	'anchor-has-content'
+];
+/**
+ * Internal dependencies
+ */
+const {
+	getTextContentFromNode,
+	getTranslateFunctionName,
+	getTranslateFunctionArgs,
+} = require( '../utils' );
+
+const meta = {
+	type: 'problem',
+	docs: {
+		description:
+			'Checks that anchors are not empty and have an href, relies on eslint-plugin-jsx-a11y/anchor-has-content under the hood. Supports usage of createInterpolateElement.',
+	},
+	fixable: 'code',
+	messages: {
+		anchorHasContent:
+			'Anchors must have content and the content must be accessible by a screen reader. Check the first parameter to createInterpolateElement.',
+		invalidMarkup:
+			'The first parameter to createInterpolateElement does not contain valid markup for an anchor.',
+	},
+	schema: [
+		{
+			type: 'object',
+			properties: {
+				components: {
+					type: 'array',
+					items: {
+						type: 'string',
+					},
+					uniqueItems: true,
+					additionalItems: false,
+				},
+			},
+		},
+	],
+};
+
+/**
+ *
+ * @param {*} tags     markup used in createInterpolateElement
+ * @param {*} tagNames names that we recognize as anchors, as configured in the eslint rules
+ * @return {Array} tagNames an array of the content of the <a> (or other) tags
+ */
+const getATagsContent = ( tags, tagNames ) => {
+	const tagText = tagNames.join( '|' );
+	const regex = new RegExp( `<(${ tagText })>(.*?)</(${ tagText })>`, 'g' );
+	const tagParts = [ ...tags.matchAll( regex ) ];
+	return tagParts.map( ( element ) => {
+		return element[ 2 ].trim();
+	} );
+};
+
+/**
+ *
+ * @param {Node} node the first param of createInterpolateElement
+ * @return {string|null} a string of the content of the node
+ */
+const getValueFromInterpolateElement = ( node ) => {
+	let nodeStr = null;
+	let args = [];
+	if ( node.type === 'CallExpression' ) {
+		const argFunctionName = getTranslateFunctionName( node.callee );
+		// get text if we are going through a translate function
+		args = getTranslateFunctionArgs(
+			argFunctionName,
+			node.arguments,
+			false
+		).map( getTextContentFromNode );
+		// An unknown function call in a translate function may produce a valid string
+		// but verifying it is not straight-forward, so we bail
+		if ( args.filter( Boolean ).length === 0 ) {
+			nodeStr = null;
```

Comment:
Why is this necessary if we override it immediately on L91?

Reply:
😬 this was just a mistake while refactoring and I didn't have a test to cover it, thanks for catching it 
Added a quick scenario to cover it https://github.com/WordPress/gutenberg/pull/50065/files#diff-4c555e1042f88bdebf5cba43ba582ae78b721c7d64c1d0cc527ea1975ef191edR61